homeassistant/components/light/tplink.py

```diff
@@ -72,7 +92,12 @@ def turn_on(self, **kwargs):
         if ATTR_BRIGHTNESS in kwargs:
             brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
             self.smartbulb.brightness = brightness_to_percentage(brightness)
-        self.smartbulb.state = self.smartbulb.BULB_STATE_ON
+        if ATTR_RGB_COLOR in kwargs:
+            rgb = kwargs.get(ATTR_RGB_COLOR)
+            self._rgb = rgb
```

Comment:
Don't assign it to the instance, none of the other attributes do that. The attribute will be updated when Home Assistant calls update after executing this method.